streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -606,6 +606,25 @@ public void onAssignment(Assignment assignment) {
             }
         }
         metadataWithInternalTopics = Cluster.empty().withPartitions(topicToPartitionInfo);
+        updateSubscribedTopics(assignment);
+    }
+
+
+    private void updateSubscribedTopics(final Assignment assignment) {
```

Comment:
nit: could we change this to `maybeUpdateSubscribedTopics(Set<String> topics)` with the check on `streamThread.builder.sourceTopicPattern() != null` within the function so that both `subscription()` and  `onAssignment` could consolidate on this function?